src/Dependencies/Collections/SegmentedArray`1.cs

```diff
@@ -94,6 +94,39 @@ public SegmentedArray(int length)
             }
         }
 
+        public SegmentedArray<T> Resize(int newLength)
```

Comment:
Add some doc comments

Reply:
This heavily deviates from expected array semantics and should not be added. There are other internal members for converting between `T[][]` and `SegmentedArray<T>` which should instead be used for crafting this specialized resize under the tight constraints of `SegmentedList<T>`.

Reply:
Would love to come up with something that fits your request here, while still sharing the duplicated segments between the old and new. Can you maybe give a bit more detail about what you are proposing?

Reply:
Lots of possibilities, first one to come to mind:

1) Move this code to PrivateMarshal
2) Change it's name to something like Grow to indicate it's differs from the standard Array.Resize mechanism
3) Modify this SegmentedArray's _items directly and not return a new SegmentedArray. This would allow the simplify this method a bit.

Or are there additional concerns that I'm not covering?

Reply:
Rather, half of what you need is there. In addition to `AsSegments`, you'll need a corresponding helper to turn segments back into the full array.

Reply:
The helpers would be mirrored after `ImmutableCollectionsMarshal`

Reply:
@sharwell -- Moved some things around to try and match what you've mentioned. What did I miss?

Reply:
➡️ Now resolved.